fenix/app/src/main/java/org/mozilla/fenix/components/toolbar/NavigationBar.kt

```diff
@@ -0,0 +1,166 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.toolbar
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Divider
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Top-level UI for displaying the navigation bar.
+ *
+ * @param actionItems A list of [ActionItem] used to populate the bar.
+ */
+@Composable
+fun NavigationBar(actionItems: List<ActionItem>) {
+    Box(
+        modifier = Modifier
+            .background(FirefoxTheme.colors.layer1)
+            .height(48.dp)
+            .fillMaxWidth(),
+    ) {
+        Divider(color = FirefoxTheme.colors.borderPrimary)
```

Comment:
We should prefer to use our design components https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/Divider.kt#28

Reply:
that's a good piece of advice, will be checking out the `compose` folder!

Reply:
@mavduevskiy We do have quite a good coverage of our design components that are also defined in this [figma](https://www.figma.com/file/ujdiReypSypu2GhRgzArea/Android-Components?type=design&node-id=1808-16937&mode=design&t=gQbzNWuKwh4Fi8IE-0). Background: We have a design system that introduced these common design components that are used throughout the application. By writing these reusable composable components that matches these design components, it provides us a single source of truth where if a component changes (text, padding, color, etc), we will only need to edit it in one place.

I would suggest familiarizing yourself with [Button](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/button) and [ListItem](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/compose/list/ListItem.kt) as probably the most common components that we will see in any screen. If you want a better overview, you can always ping @MozillaNoah or myself to dive deeper.